src/awkward/_v2/operations/reducers/ak_all.py

```diff
@@ -9,53 +9,50 @@
-#     if all lists at a given dimension have the same length and no None values,
-#     but it generalizes to cases where they do not.
-
-#     See #ak.sum for a more complete description of nested list and missing
-#     value (None) handling in reducers.
-#     """
-#     layout = ak._v2.operations.convert.to_layout(
-#         array, allow_record=False, allow_other=False
-#     )
-#     if axis is None:
-
-#         def reduce(xs):
-#             if len(xs) == 1:
-#                 return xs[0]
-#             else:
-#                 return xs[0] and reduce(xs[1:])
-
-#         return reduce(
-#             [ak.nplike.of(x).all(x) for x in ak._v2._util.completely_flatten(layout)]
-#         )
-#     else:
-#         behavior = ak._v2._util.behaviorof(array)
-#         return ak._v2._util.wrap(
-#             layout.all(axis=axis, mask=mask_identity, keepdims=keepdims), behavior
-#         )
+    """
+    Args:
+        array: Data to combine with "logical all."
+        axis (None or int): If None, combine all values from the array into
+            a single scalar result; if an int, group by that axis: `0` is the
+            outermost, `1` is the first level of nested lists, etc., and
+            negative `axis` counts from the innermost: `-1` is the innermost,
+            `-2` is the next level up, etc.
+        keepdims (bool): If False, this reducer decreases the number of
+            dimensions by 1; if True, the reduced values are wrapped in a new
+            length-1 dimension so that the result of this operation may be
+            broadcasted with the original array.
+        mask_identity (bool): If True, reducing over empty lists results in
+            None (an option type); otherwise, reducing over empty lists
+            results in the operation's identity.
+
+    Returns True in each group of elements from `array` (many types supported,
+    including all Awkward Arrays and Records) if all values are True; False
+    otherwise. Thus, it represents reduction over the "logical and" operation,
+    whose identity is True (i.e. asking if all the values are True in an
+    empty list results in True). This operation is the same as NumPy's
+    [all](https://docs.scipy.org/doc/numpy/reference/generated/numpy.all.html)
+    if all lists at a given dimension have the same length and no None values,
+    but it generalizes to cases where they do not.
+
+    See #ak.sum for a more complete description of nested list and missing
+    value (None) handling in reducers.
+    """
+    layout = ak._v2.operations.convert.to_layout(
+        array, allow_record=False, allow_other=False
+    )
+    if axis is None:
+
+        def reduce(xs):
+            if len(xs) == 1:
+                return xs[0]
+            else:
+                return xs[0] and reduce(xs[1:])
+
+        return reduce(
+            [ak.nplike.of(x).all(x) for x in ak._v2._util.completely_flatten(layout)]
+        )
```

Comment:
Actually, I had been planning to work on these reducers, though you're more than welcome to do it. I'll find something else that needs doing. But, something I had in mind about all of these reducers: you might want to create a helper function to handle the `axis=None` case because it's very repetitive; there's a lot of duplicated code (which is now less evident because each of these functions is in a different file). The helper function can live in `ak._v2._util`, where all of these functions can see it.

Reply:
Thanks! I don't mind if you take over the branch - I will not be working on it during the weekend. I'll see were we stand on Monday. :-)

Reply:
I don't entirely know what you were going to do with the datetimes/timedeltas—is there kernel-level work needed? If so, I'll just do the part I was intending, which was the high-level interface. I can get the high-level ArrayBuilder updated, too. By Monday, I'll leave this in a state where it's only needing the datetimes/timedeltas and tests.

Reply:
Thanks! I don't think any kernel-level work is needed.